Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -2227,13 +2232,51 @@ void FunctionDefinitionWriter::visit(AST::PhonyAssignmentStatement& statement)
     m_stringBuilder.append(")");
 }
 
+static std::optional<std::pair<String, String>> fragDepthIdentifierForFunction(AST::Function* function)
+{
+    if (!function || function->stage() != ShaderStage::Fragment)
```

Comment:
@tadeuzagallo this seems to work and should be local to the current function, what do you think?

Reply:
Looks good. I was concerned if frag_depth could be nested further in another struct, but it seems like that's not allowed.